src/Sylius/Bundle/ApiBundle/Provider/CustomerProvider.php

```diff
@@ -30,17 +33,33 @@ final class CustomerProvider implements CustomerProviderInterface
     /** @var CustomerRepositoryInterface */
     private $customerRepository;
 
+    /** @var UserContextInterface */
+    private $userContext;
+
     public function __construct(
         CanonicalizerInterface $canonicalizer,
         FactoryInterface $customerFactory,
-        CustomerRepositoryInterface $customerRepository
+        CustomerRepositoryInterface $customerRepository,
+        UserContextInterface $userContext
     ) {
         $this->canonicalizer = $canonicalizer;
         $this->customerFactory = $customerFactory;
         $this->customerRepository = $customerRepository;
+        $this->userContext = $userContext;
+    }
+
+    public function get(): ?CustomerInterface
+    {
+        /** @var UserInterface|null $user */
+        $user = $this->userContext->getUser();
+        if ($user !== null && $user instanceof ShopUserInterface) {
+            return $user->getCustomer();
+        }
+
+        return null;
     }
 
-    public function provide(string $email): CustomerInterface
+    public function getOrCreate(string $email): CustomerInterface
```

Comment:
You changed function in service, it should be mentioned in the upgrade file (even if it is `experimental`)

Reply:
I would be for reverting this change